torch/onnx/ops/_impl.py

```diff
@@ -0,0 +1,317 @@
+        assert len(encoded.attr_keys) == len(encoded.attr_types), (
+            f"Mismatch between number of attribute keys and types: {len(encoded.attr_keys)} != {len(encoded.attr_types)}"
+        )
+        assert len(encoded.attr_keys) == len(encoded.attr_pos), (
+            f"Mismatch between number of attribute keys and positions: {len(encoded.attr_keys)} != {len(encoded.attr_pos)}"
+        )
+        return encoded
+
+    def to_dict(self) -> dict[str, Union[int, float, str]]:
+        attrs = {}
+        for i, key in enumerate(self.attr_keys):
+            attr_type = self.attr_types[i]
+            if attr_type == "i":
+                attrs[key] = self.attr_ints[self.attr_pos[i][0]]
+            elif attr_type == "f":
+                attrs[key] = self.attr_floats[self.attr_pos[i][0]]
+            elif attr_type == "s":
+                attrs[key] = self.attr_strs[self.attr_pos[i][0]]
+            elif attr_type == "t":
+                attrs[key] = self.attr_tensors[self.attr_pos[i][0]]
+            elif attr_type == "fs":
+                attrs[key] = self.attr_floats[self.attr_pos[i][0] : self.attr_pos[i][1]]
+            elif attr_type == "is":
+                attrs[key] = self.attr_ints[self.attr_pos[i][0] : self.attr_pos[i][1]]
+            elif attr_type == "ss":
+                attrs[key] = self.attr_strs[self.attr_pos[i][0] : self.attr_pos[i][1]]
+            elif attr_type == "ts":
+                attrs[key] = self.attr_tensors[
+                    self.attr_pos[i][0] : self.attr_pos[i][1]
+                ]
+            else:
+                raise ValueError(f"Unsupported attribute type: {attr_type}")
+        return attrs
+
+
+@torch.library.custom_op(
+    "onnx_symbolic::_symbolic",
+    mutates_args=(),
+    schema="(Tensor[] inputs, str op_type, int onnx_dtype, Tensor[] attr_tensors, *, SymInt[] shape, str[] attr_keys, str[] attr_types, int[][] attr_pos, int[] attr_ints, float[] attr_floats, str[] attr_strs, str[] metadata_props_keys, str[] metadata_props_values, str domain='', int? version=None) -> Tensor",
+)
+def _symbolic(
+    inputs: Sequence[torch.Tensor],
+    op_type: str,
+    onnx_dtype: int,
+    attr_tensors: Sequence[torch.Tensor],
+    *,
+    shape: Sequence[Union[int, torch.SymInt]],
+    attr_keys: Sequence[str],
+    attr_types: Sequence[str],
+    attr_pos: Sequence[tuple[int, int]],
+    attr_ints: Sequence[int],
+    attr_floats: Sequence[float],
+    attr_strs: Sequence[str],
+    metadata_props_keys: Sequence[str] = (),
+    metadata_props_values: Sequence[str] = (),
+    domain: str = "",
+    version: Optional[int] = None,
+) -> torch.Tensor:
+    # TODO: Verify that shape supports SymInt
+    torch._check(
+        onnx_dtype in _ONNX_DTYPE_TO_TORCH_DTYPE,
+        f"{onnx_dtype} is invalid as an ONNX data type. Valid values are {list(_ONNX_DTYPE_TO_TORCH_DTYPE.keys())}",
+    )
+    return torch.zeros(shape, dtype=_ONNX_DTYPE_TO_TORCH_DTYPE[onnx_dtype])
+
+
+@_symbolic.register_fake
```

Comment:
Could you link a ref here. I just wanted to learn more about this.

Reply:
Sure: https://pytorch.org/docs/stable/library.html#torch.library.register_fake